torch/_inductor/lowering.py

```diff
@@ -3768,56 +3768,66 @@ def _validate_reduction_axis(x, axis):
     return axis
 
 
-def make_reduction(reduction_type: str, override_return_dtype=None):
-    def inner(x, axis=None, keepdims=False, *, dtype=None):
-        if dtype is not None:
-            x = to_dtype(x, dtype)
-        size = x.get_size()
-        axis = set(_validate_reduction_axis(x, axis))
-
-        kept_sizes = []
-        kept_idx = []
-        reduced_sizes = []
-        reduced_idx = []
-        for i in range(len(size)):
-            if i in axis:
-                reduced_idx.append(i)
-                reduced_sizes.append(size[i])
-            else:
-                kept_idx.append(i)
-                kept_sizes.append(size[i])
-
-        def loader(index, reduction_index):
-            assert len(reduction_index) == len(reduced_idx)
-            if keepdims:
-                assert len(index) == len(size)
-                assert all(index[i] == 0 for i in reduced_idx)
-                index = [index[i] for i in kept_idx]
-            assert len(index) == len(kept_idx)
-            new_index = [None] * (len(index) + len(reduction_index))
-            for idx, var in itertools.chain(
-                zip(kept_idx, index), zip(reduced_idx, reduction_index)
-            ):
-                new_index[idx] = var
-            return inner_loader(new_index)
+def _make_reduction_inner(x, *, axis, keepdims, dtype, override_return_dtype):
```

Comment:
why are the diffs so bad :'(

Reply:
Is this part all just code shuffling?

Reply:
The diff is much clearer if you ignore whitespace changes,
https://github.com/pytorch/pytorch/pull/102486/files?diff=unified&w=1

Reply:
TIL!